src/mds/QuiesceDbManager.cc

```diff
@@ -111,6 +109,8 @@ void* QuiesceDbManager::quiesce_db_thread_main()
           next_event_at_age = leader_upkeep(std::move(acks), std::move(requests));
         } else {
           // not yet there. Put the acks and requests back onto the queue and wait for updates
+          // We should mark the next event age in case we get caught up in the sleep above
+          next_event_at_age = db.get_age() + bootstrap_delay;
```

Comment:
I'm finding it hard to understand this, where is this modified val being used? `next_event_at_age` is being updated in the else blocks, and then the `QuiesceDbManager::quiesce_db_thread_main()` doesn't make use of it.

Reply:
another (dumb) question, if the `next_event_at_age` is already max(), we're adding some duration to it here?

Reply:
`next_event_at_age` is used [here](https://github.com/ceph/ceph/pull/57946/files/b3edbbabdbf64c39fb91da3f47fcfce25bd3c93c#diff-f5065f0feaead10f74178b07ecb08b30d186380ef35684c0364024fe42f62a04R87) and [here](https://github.com/ceph/ceph/pull/57946/files/b3edbbabdbf64c39fb91da3f47fcfce25bd3c93c#diff-f5065f0feaead10f74178b07ecb08b30d186380ef35684c0364024fe42f62a04R91). It will cause the main thread to make a pass at or after the designated age.

This var is initialized with `max()` to denote "no time-based events are pending", which will make the main thread sleep until the submit condition is triggered.
However, in the body of the main thread loop the `next_event_at_age` is updated with the output of the various subroutines, e.g. at the line you commented on.

Reply:
If the db thread has some work, and the boostrap_interval is non-zero, what happens in that case?